src/HotChocolate/Fusion-vnext/src/Fusion.Composition/SourceSchemaPreProcessor.cs

```diff
@@ -0,0 +1,102 @@
+using HotChocolate.Fusion.Extensions;
+using HotChocolate.Skimmed;
+using HotChocolate.Types;
+
+namespace HotChocolate.Fusion;
+
+/// <summary>
+/// Applies @lookup, @key and optionally @shareable to a source schema to make it equivalent to a Fusion v1 source schema.
+/// </summary>
+internal class SourceSchemaPreProcessor(SchemaDefinition schemaDefinition, SourceSchemaPreProcessorOptions? options = null)
```

Comment:
I'm mutating the original SchemaDefinition. Is this fine or should I implement some cloning mechanism, so the preprocessor doesn't change the original SchemaDefinition?

Reply:
This is fine